server/internal/mcp/impl.go

```diff
@@ -380,12 +383,61 @@ func (s *Service) ServePublic(w http.ResponseWriter, r *http.Request) error {
 				Token:        externalSecret.Token,
 			})
 		}
+	case !toolset.McpIsPublic && toolset.OauthProxyServerID.Valid:
```

Comment:
I still think we need to find a way to get more specific for the oauth piece. Ideally we should find a way to look at the actual oauth provider type provider == "gram" then we run the gram oauth logic

If we need to query for more information that is no problem